internal/grpc/services/gateway/storageprovider.go

```diff
@@ -510,7 +509,7 @@ func (s *svc) InitiateFileDownload(ctx context.Context, req *provider.InitiateFi
 
 			// TODO(labkode): calculate signature of the whole request? we only sign the URI now. Maybe worth https://tools.ietf.org/html/draft-cavage-http-signatures-11
 			target := u.String()
-			token, err := s.sign(ctx, target)
+			token, err := s.sign(ctx, target, s.c.TransferExpires)
```

Comment:
That seems wrong. `s.c.TransferExpires` is an amount of seconds, while expiresAt requires a timestamp.

Speaking of which: Why did we make expiresAt an paramter to `sign()` instead of just calculating it there?

Reply:
Fixed. It became a parameter because `expiresAt` can - and in fact generally will be - set to something different than the gateway fallback expiration duration.